spynnaker/pyNN/models/neuron/abstract_population_vertex.py

```diff
@@ -262,17 +264,36 @@ def _get_sdram_usage_for_neuron_params(self, vertex_slice):
 
     def _get_sdram_usage_for_atoms(
             self, vertex_slice, graph, machine_time_step):
-        sdram_requirement = (
-            SYSTEM_BYTES_REQUIREMENT +
-            self._get_sdram_usage_for_neuron_params(vertex_slice) +
-            self._neuron_recorder.get_static_sdram_usage(vertex_slice) +
-            PopulationMachineVertex.get_provenance_data_size(
-                PopulationMachineVertex.N_ADDITIONAL_PROVENANCE_DATA_ITEMS) +
-            self.__synapse_manager.get_sdram_usage_in_bytes(
-                vertex_slice, machine_time_step, graph, self) +
-            profile_utils.get_profile_region_size(self.__n_profile_samples))
-
-        return sdram_requirement
+
+        # These values can be cross checked with the data_spec_text_files
+        # SYSTEM = 0
+        region_0 = SIMULATION_N_BYTES
+        # NEURON_PARAMS = 1
+        region_1 = self._get_sdram_usage_for_neuron_params(vertex_slice)
+        # NEURON_RECORDING = 6
+        region_6 = self._neuron_recorder.get_data_spec_sdram_usage(
+            vertex_slice)
+        # PROVENANCE_DATA = 7
+        region_7 = PopulationMachineVertex.get_provenance_data_size(
+                PopulationMachineVertex.N_ADDITIONAL_PROVENANCE_DATA_ITEMS)
+        # SYNAPSE_PARAMS = 2
+        # POPULATION_TABLE = 3
+        # SYNAPTIC_MATRIX = 4
+        # SYNAPSE_DYNAMICS = 5
+        # DIRECT_MATRIX = 10
+        sm_regions = self.__synapse_manager.get_sdram_usage_in_bytes(
+                vertex_slice, machine_time_step, graph, self)
+        # PROFILING = 8
+        region_8 = profile_utils.get_profile_region_size(
+            self.__n_profile_samples)
+
+        overhead = APP_PTR_TABLE_HEADER_BYTE_SIZE + MAX_MEM_REGIONS * 4
```

Comment:
Just saw this too; there is a constant for this in FEC called DATA_SPECABLE_BASIC_SETUP_INFO_N_BYTES.  This also adds a sark allocation, so you might not need this below if this constant is used.

Reply:
I intentional did not use DATA_SPECABLE_BASIC_SETUP_INFO_N_BYTES because it had the spark allocation included.  Removing it allowed me to come up with a number that could be compared with HostExecuteDataSpecification.__malloc_region_storage methods size param.

Reply:
Ah OK - in that case the equivalent constant is APP_PTR_TABLE_BYTE_SIZE (which is from data_specification.constants)